x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_list.tsx

```diff
@@ -0,0 +1,226 @@
+}: {
+  crumbs: Array<{ text: string | JSX.Element; onClick: () => void }>;
+  matchingEventEntries: MatchingEventEntry[];
+}) {
+  return (
+    <>
+      <StyledBreadcrumbs breadcrumbs={crumbs} />
+      <EuiSpacer size="l" />
+      <>
+        {matchingEventEntries.map((eventView, index) => {
+          return (
+            <Fragment key={index}>
+              <EuiText>
+                <BoldCode>{eventView.eventType}</BoldCode>
+                {' @ '}
+                {eventView.formattedDate}
+              </EuiText>
+              <EuiSpacer size="xs" />
+              <EuiButtonEmpty onClick={eventView.setQueryParams}>{eventView.name}</EuiButtonEmpty>
+              {index === matchingEventEntries.length - 1 ? null : <EuiHorizontalRule margin="m" />}
+            </Fragment>
+          );
+        })}
+      </>
+    </>
+  );
+});
+
+export const ProcessEventListNarrowedByType = memo(function ProcessEventListNarrowedByType({
+  processEvent,
+  eventType,
+  relatedStats,
+  pushToQueryParams,
+}: {
+  processEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  eventType: string;
+  relatedStats: ResolverNodeStats;
+}) {
+  const processName = processEvent && event.eventName(processEvent);
+  const processEntityId = event.entityId(processEvent);
+  const totalCount = Object.values(relatedStats.events.byCategory).reduce(
+    (sum, val) => sum + val,
+    0
+  );
+  const eventsString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.events',
+    {
+      defaultMessage: 'Events',
+    }
+  );
+  const waitingString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.wait',
+    {
+      defaultMessage: 'Waiting For Events...',
+    }
+  );
+
+  const relatedsReadyMap = useSelector(selectors.relatedEventsReady);
+  const relatedsReady = relatedsReadyMap.get(processEntityId);
+
+  const relatedEventsForThisProcess = useSelector(selectors.relatedEventsByEntityId).get(
+    processEntityId
+  );
+  const dispatch = useResolverDispatch();
+
+  useEffect(() => {
+    if (typeof relatedsReady === 'undefined') {
+      dispatch({
+        type: 'userRequestedRelatedEventData',
+        payload: processEntityId,
+      });
+    }
+  }, [relatedsReady, dispatch, processEntityId]);
+
+  const waitCrumbs = useMemo(() => {
+    return [
+      {
+        text: eventsString,
+        onClick: () => {
+          pushToQueryParams({ crumbId: '', crumbEvent: '' });
+        },
+      },
+    ];
+  }, [pushToQueryParams, eventsString]);
+
+  const relatedEventsToDisplay = useMemo(() => {
+    return relatedEventsForThisProcess?.events || [];
+  }, [relatedEventsForThisProcess?.events]);
+
+  /**
+   * A list entry will be displayed for each of these
+   */
+  const matchingEventEntries: MatchingEventEntry[] = useMemo(() => {
+    const relateds = relatedEventsToDisplay
+      .reduce((a: ResolverEvent[], candidate) => {

```

Comment:
let's discuss moving this logic to a selector

Reply:
Kinda goes with the comment I made above: I generally agree that we should move things to selectors. I'm just curious "how much" in light of things you've expressed in the past regarding avoiding gumming them up with UI-derived things like queryParams.

Reply:
let's sync in person.